streams/src/test/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImplTest.java

```diff
@@ -258,8 +283,14 @@ public void shouldNotAcceptNullStoreNameWhenReducingSessionWindows() throws Exce
         groupedStream.reduce(MockReducer.STRING_ADDER, SessionWindows.with(10), (String) null);
     }
 
+    @Test(expected = InvalidTopicException.class)
+    public void shouldNotAcceptInvalidStoreNameWhenReducingSessionWindows() throws Exception {
+        String storeName = "~foo bar~";
+        groupedStream.reduce(MockReducer.STRING_ADDER, SessionWindows.with(10), storeName);
+    }
+
     @Test(expected = NullPointerException.class)
-    public void shouldNotAcceptNullStateStoreSupplierNameWhenReducingSessionWindows() throws Exception {
+    public void shouldNotAcceptNullStateStoreSupplierWhenReducingSessionWindows() throws Exception {
```

Comment:
I did a couple of cleanups like this in the cases where it seemed like the test name did not match the actual test, since I am technically writing and editing tests regarding the state store names.